src/MSBuildProjectBuilder/ProjectBuilder.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.Build.Construction;
+using System.Collections.Generic;
+
+namespace Microsoft.MSBuildProjectBuilder
+{
+    public partial class ProjectBuilder
+    {
+        public ProjectRootElement ProjectRoot { get; private set; }
+
+        private ICollection<ProjectElement> lastElements = new List<ProjectElement>();
```

Comment:
I think this makes the class harder to understand.  Perhaps it would be better to have separate `lastItemGroup`, `lastPropertyGroup`, etc.  Do you feel that the collection made things more complex?